### PR TITLE
EOS-16238: Update s3_setup script to read confstore config file URL as cmdline argument.

### PR DESCRIPTION
**Pull request by amitwac2608**
Also update setupReplicationScript.sh to read host names from a comm-separated string,
instead of a text file.

Signed-off-by: Amit Kumar <amit.kumar@seagate.com>

### CODE REVIEW

File: scripts/provisioning/s3_setup [hunk 1]
```diff
@@ -247,58 +230,10 @@ then
 setup_release_repo
 fi
 
-# Install required S3 release rpm dependencies.
```
Comment (amitwac2608) [inline scripts/provisioning/s3_setup:250]: removed this as this section seems to be obsolete. Need review from @shalakadharap , @AjinkyaDhumal , @sachitanands

File: scripts/provisioning/s3_setup [hunk 2]
```diff
@@ -121,17 +107,8 @@ do
     --purgeconfigldap )
         purge_config_ldap=true
         ;;
-    --ldapreplication )
```
Comment (amitwac2608) [inline scripts/provisioning/s3_setup:124]: these options seems obsolete now. Need review from @shalakadharap  ans @sachitanands

**Review by shalakadharap — APPROVED** [review]
Approving setupLdapReplication script related changes.
Not really sure on s3_setup script changes.
Check with @bkirunge7  1s.

File: s3cortxutils/s3confstore/s3confstore/cortx_s3_confstore.py [hunk 3]
```diff
@@ -33,7 +33,11 @@ def __init__(self):
 
   def load_config(self, index: str, backend_url: str):
     """Load Config into constore"""
-    Conf.load(index, backend_url)
+    if not backend_url.strip():
+      print("backend_url is empty")
```
Comment (bkirunge7) [inline s3cortxutils/s3confstore/s3confstore/cortx_s3_confstore.py:37]: can this be more verbose, and help in debugging issue problem. print as much info or any variable values as possible.

File: scripts/ldap/replication/setupReplicationScript.sh [hunk 4]
```diff
@@ -45,19 +45,22 @@ fi
...
+# checkHostValidity will check if all provided hosts are valid and reachable
 checkHostValidity()
 {
     while read host; do
-        isValid=`ping -c 1 ${host} | grep bytes | wc -l`
+        isValid=$(ping -c 1 ${host} | grep bytes | wc -l)
         if [ "$isValid" -le 1 ]
         then
-            echo ${host}" is either invalid or not reachable. Please check or correct your entry in host file"
+            echo "$host is either invalid or not reachable."
```
Comment (bkirunge7) [inline scripts/ldap/replication/setupReplicationScript.sh:55]: also print isValid value

File: scripts/provisioning/s3_setup [hunk 5]
```diff
@@ -330,18 +272,20 @@ if [[ $config_ldap == true ]]
...
-    IFS=','
-    read -ra hosts_arr <<< "$cluster_hosts"
-    if [ ${#hosts_arr[@]} -gt 1 ]
+    # check number of nodes in the cluster
+    server_nodes_count=$(s3confstore --getkey "cluster>node_count" --path "$confstore_config_url")
+    if [ "$server_nodes_count" -gt 1 ]
     then
+        for ((index=0; index<"$server_nodes_count"; index++))
+        do
+            host=$(s3confstore --getkey "cluster>server["$index"]>hostname" --path "$confstore_config_url")
```
Comment (bkirunge7) [inline scripts/provisioning/s3_setup:281]: can we catch error if s3confstore fails

Comment (amitwac2608) [reply scripts/provisioning/s3_setup:281]: will add check if s3onfstore failed or not.

File: scripts/ldap/replication/setupReplicationScript.sh (hunk 4 above)
Comment (amitwac2608) [reply scripts/ldap/replication/setupReplicationScript.sh:55]: ok.

File: s3cortxutils/s3confstore/s3confstore/cortx_s3_confstore.py (hunk 3 above)
Comment (amitwac2608) [reply s3cortxutils/s3confstore/s3confstore/cortx_s3_confstore.py:37]: I will ad more information.

**Review by bkirunge7 — APPROVED** [review]

**Comment by amitwac2608** [comment]
Jenkins job on latest commit hash: http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/396

PASSED

**Pull request merged**